sync-core/src/main/java/com/cloudant/sync/sqlite/SQLDatabaseFactory.java

```diff
@@ -36,30 +36,47 @@
 
     private final static Logger logger = Logger.getLogger(SQLDatabaseFactory.class.getCanonicalName());
 
-    public static SQLDatabase createSQLDatabase(String dbFilename) throws IOException {
+    /**
+     * SQLCipher-based implementation for creating database.
+     * @param dbFilename full file path of the db file
+     * @param provider Key provider object storing the SQLCipher key
+     *                 Supply a NullKeyProvider to use a non-encrypted database.
+     * @return {@code SQLDatabase} for the given filename
+     * @throws IOException if the file does not exists, and also
+     *         can not be created
+     */
+    public static SQLDatabase createSQLDatabase(String dbFilename, KeyProvider provider) throws IOException {
+
+        boolean runningOnAndroid =  Misc.isRunningOnAndroid();
+        boolean useSqlCipher = (provider.getEncryptionKey() != null);
 
         makeSureFileExists(dbFilename);
-        if(Misc.isRunningOnAndroid()) {
-            try {
-                Class c = Class.forName("com.cloudant.sync.sqlite.android.AndroidSQLite");
 
-                Method m = c.getMethod("createAndroidSQLite",String.class);
-                return (SQLDatabase)m.invoke(null,dbFilename);
+        try {
 
-            } catch (Exception e) {
-                logger.log(Level.SEVERE, "Failed to load database module", e);
-                return null;
-            }
-        } else {
-            try {
-                Class c = Class.forName("com.cloudant.sync.sqlite.sqlite4java.SQLiteWrapper");
-                Method m = c.getMethod("openSQLiteWrapper", String.class);
-                return (SQLDatabase)m.invoke(null, dbFilename);
-
-            } catch (Exception e) {
-                logger.log(Level.SEVERE,"Failed to load database module",e);
-                return null;
+            if (runningOnAndroid) {
+                if (useSqlCipher) {
+                    return (SQLDatabase) Class.forName("com.cloudant.sync.sqlite.android.AndroidSQLCipherSQLite")
+                            .getMethod("createAndroidSQLite", String.class, KeyProvider.class)
+                            .invoke(null, new Object[]{dbFilename, provider});
+                } else {
+                    return (SQLDatabase) Class.forName("com.cloudant.sync.sqlite.android.AndroidSQLite")
+                            .getMethod("createAndroidSQLite", String.class)
+                            .invoke(null, dbFilename);
+                }
+            } else {
+                if (useSqlCipher) {
+                    throw new IOException("No SQLCipher-based database implementation for Java SE");
```

Comment:
I added the UOE in https://github.com/cloudant/sync-android/commit/f9854b4f3e11f5b719e7833d9dbcd574267714a3